sjsonnet/src-jvm/sjsonnet/SjsonnetMain.scala

```diff
@@ -105,15 +106,25 @@ object SjsonnetMain {
       config.preserveOrder
     )
 
-    def writeFile(f: os.RelPath, contents: String): Either[String, Unit] = {
-      Try(os.write.over(os.Path(f, wd), contents, createFolders = config.createDirs))
-        .toEither
-        .left
-        .map{
-          case e: NoSuchFileException => s"open $f: no such file or directory"
-          case e => e.toString
-        }
-    }
+    def handleWriteFile[T](f: => T): Either[String, T] =
+      Try(f).toEither.left.map{
+        case e: NoSuchFileException => s"open $f: no such file or directory"
+        case e => e.toString
+      }
+
+    def writeFile(f: os.RelPath, contents: String): Either[String, Unit] =
+      handleWriteFile(os.write.over(os.Path(f, wd), contents, createFolders = config.createDirs))
+
+    def writeToFile[U](f: os.RelPath, contents: Writer => Either[String, U]): Either[String, Unit] =
+      handleWriteFile(os.write.over.outputStream(os.Path(f, wd), createFolders = config.createDirs)).flatMap { out =>
+        try {
+          val buf = new BufferedOutputStream(out)
+          val wr = new OutputStreamWriter(buf, StandardCharsets.UTF_8)
+          val u = contents(wr)
+          wr.flush()
```

Comment:
Do we need to call `buf.flush()` here?

Reply:
No, flushing (like closing) is transitive, you only need to call it once for the first stream/writer in the pipeline.